lib/mj-single.js

```diff
@@ -549,6 +551,20 @@ function GetSpeech(result) {
 }
 
 //
+//  Creates the semantic tree for the current element and attaches it as JSON
+//  and XML.
+//
+function GetSemantic(result) {
+  if (!data.semantic) return;
+  if (!result.mml) {
+    data.mml = true ;
+    GetMML(result);
```

Comment:
Since there is not much more cleaning up to do (as far as I can see), I put the delete result.mml into the ReturnResult function now.